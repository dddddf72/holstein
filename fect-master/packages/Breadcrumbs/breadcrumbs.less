.fect-nav_container {
  margin: 0;
  padding: 0;
  line-height: inherit;
  color: var(--accents-4);
  box-sizing: border-box;
  display: flex;
  align-items: center;
}

.fect-nav_container {
  .fect-bread_separator {
    display: inline-flex;
    margin: 0 8px;
    user-select: none;
    pointer-events: none;
    align-items: center;
  }
  .fect-braed_item {
    &:last-child {
      .fect-bread_separator {
        display: none;
      }
    }
  }
}
