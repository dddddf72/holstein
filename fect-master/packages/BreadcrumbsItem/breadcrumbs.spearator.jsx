import { defineComponent } from 'vue'

const Separator = defineComponent({
  setup(props, { slots }) {
    return () => <div className="fect-bread_separator">{slots.default?.()}</div>
  },
})

export default Separator
