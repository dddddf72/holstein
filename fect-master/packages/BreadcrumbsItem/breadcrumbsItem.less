.fect-braed_item {
  display: inline-flex;
  align-items: center;
  user-select: none;
  cursor: pointer;
  &:last-child {
    color: var(--accents-6);
  }
  .withLink {
    font-weight: 550;
  }
}
