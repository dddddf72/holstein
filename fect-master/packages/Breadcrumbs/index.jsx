import { computed } from 'vue'
import { createNameSpace, theme, validator, createProvider } from '../utils'
const [createComponent] = createNameSpace('Breadcrumbs')
import './breadcrumbs.less'

const { normalSizes } = theme

const READONLY_BREADCRUMBS_KEY = 'breadcrumbsKey'

const queryFontSize = (size) => {
  const sizesPool = {
    mini: '12px',
    small: '14px',
    medium: '16px',
    large: '18px',
  }
  return sizesPool[size]
}

export default createComponent({
  props: {
    separator: {
      type: String,
      default: '/',
    },
    size: {
      type: String,
      default: 'medium',
      validator: validator.enums(normalSizes),
    },
  },
  setup(props, { slots }) {
    const { provider } = createProvider(READONLY_BREADCRUMBS_KEY)
    provider({ separator: props.separator })

    const safeSize = computed(() => {
      const styles = {
        fontSize: queryFontSize(props.size),
      }
      return styles
    })

    return () => (
      <nav class="fect-nav_container" style={safeSize.value}>
        {slots.default?.()}
      </nav>
    )
  },
})
