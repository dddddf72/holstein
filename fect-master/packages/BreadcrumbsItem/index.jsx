import { ref, watchEffect } from 'vue'
import { createNameSpace, useProvider } from '../utils'
import Link from '../Link'
const [createComponent] = createNameSpace('BreadcrumbsItem')
import './breadcrumbsItem.less'
import Separator from './breadcrumbs.spearator'
const READONLY_BREADCRUMBS_KEY = 'breadcrumbsKey'

export default createComponent({
  props: {
    to: {
      type: [String, Object],
      default: () => {},
    },
    href: {
      type: String,
      default: '',
    },
  },
  setup(props, { slots, attrs }) {
    const hasLink = ref(false)

    watchEffect(() => {
      /**
       * use Boolean to  display ,magic implicit convert is no encourage
       */
      const isNull = Boolean(props?.to) || props.href !== ''
      hasLink.value = isNull
    })
    const { ctx } = useProvider(READONLY_BREADCRUMBS_KEY)

    if (!ctx) {
      if (process.env.NODE_ENV !== 'production') {
        console.error(
          '[Fect] <BreadcrumbsItem> must be a child component of <Breadcrumbs>.',
        )
      }
      return
    }

    const content = slots.default?.()

    const withoutLinkRender = () => {
      return (
        <span class="fect-braed_item">
          {content}
          <Separator>{ctx?.separator}</Separator>
        </span>
      )
    }
    const linkRender = () => {
      return (
        <div class="fect-braed_item ">
          <Link
            to={props.to}
            href={props.href}
            class="withLink"
            target={attrs.target}
          >
            {content}
          </Link>
          <Separator>{ctx?.separator}</Separator>
        </div>
      )
    }

    return () => (
      <>{hasLink.value ? <>{linkRender()}</> : <>{withoutLinkRender()}</>}</>
    )
  },
})
