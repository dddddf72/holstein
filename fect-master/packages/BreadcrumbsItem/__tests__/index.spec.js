import { mount } from '@vue/test-utils'
import BreadcrumbsItem from '../index'
import Breadcrumbs from '../../Breadcrumbs'

describe('BreadcrumbsItem', () => {
  it('should render as element', () => {
    const breadcrumbsItem = mount(
      <Breadcrumbs>
        <BreadcrumbsItem />
      </Breadcrumbs>,
    )
    expect(() => breadcrumbsItem.unmount()).not.toThrow()
  })
  it('when component have not to or href,it should render as span', () => {
    const wrapper = mount(
      <Breadcrumbs>
        <BreadcrumbsItem />
      </Breadcrumbs>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.fect-braed_item').html()).toBe(
      '<span class="fect-braed_item"><!----><div class="fect-bread_separator">/</div></span>',
    )
  })
})
