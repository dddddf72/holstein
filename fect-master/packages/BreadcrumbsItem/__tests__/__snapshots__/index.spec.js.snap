// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbsItem when component have not to or href,it should render as span 1`] = `<nav class="fect-nav_container" style="font-size: 16px;"><span class="fect-braed_item"><!----><div class="fect-bread_separator">/</div></span></nav>`;
